SerilogAnalyzer/SerilogAnalyzer/SerilogAnalyzerPascalCaseCodeFixProvider.cs

```diff
@@ -0,0 +1,74 @@
+namespace SerilogAnalyzer
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SerilogAnalyzerCodeFixProvider)), Shared]
+    public class SerilogAnalyzerPascalCaseCodeFixProvider : CodeFixProvider
+    {
+        private const string title = "Pascal case the property";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(SerilogAnalyzerAnalyzer.PascalPropertyNameDiagnosticId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var declaration = root.FindNode(diagnosticSpan);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    title: title,
+                    createChangedSolution: c => this.PascalCaseTheProperties(context.Document, (ArgumentSyntax)declaration, c),
+                    equivalenceKey: title),
+                diagnostic);
+        }
+
+        private async Task<Solution> PascalCaseTheProperties(Document document, ArgumentSyntax node, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var oldToken = node.GetFirstToken();
+
+            var text = oldToken.ValueText;
+            var matches = Regex.Matches(text, "{@?[a-z]\\w+?}");
```

Comment:
Besides `@` we need to take care of the other possible prefix character `$` which is used for forced stringification.
Does this also replace escaped formats? e.g. `{{hello world}}`